frontend/front/src/components/SurveyComponent/SurveyComponent.js

```diff
@@ -11,217 +17,230 @@ import { useNavigate } from "react-router-dom";
 /*
  * Reusable survey component based on https://docs.google.com/document/d/1LPCNCUBJR8aOCEnO02x0YG3cPMg7CzThlnDzruU1KvI/edit
  */
-export const SurveyComponent = ({
-  submitSurvey,
-  isLoading,
-  activeHome,
-  isEditable,
-  surveyId,
-  formSpacing,
-  defaultData,
-  onDelete,
-}) => {
-  const navigate = useNavigate();
+export const SurveyComponent = forwardRef(
```

Comment:
Wrapping this in forwardRef caused the diff to be ridiculous - most of the code is the same here